### ISSUE TITLE
Support more text styles

### ISSUE THREAD

**Issue by GoogleCodeExporter** [garglk/garglk#201]
```
I was surprised to see that Gargoyle doesn't support changing the text 
justification or text size. For an interpreter that claims to care about 
typography this is sorely missed!
```

Original issue reported on code.google.com by `curiousdannii` on 1 Feb 2013 at 5:59

**Comment by alysdexia** [comment]
not a defect

**Comment by Tynach** [comment]
@angstsmurf, using your test gblorb file on #338, Gargoyle currently shows this:

![Screenshot_20210617_115424](https://user-images.githubusercontent.com/224197/122456944-f6d79400-cf62-11eb-8e2d-0f5d4c734250.png)

Does the 'a' in 'Chiroplasty' use a style synonymous with either user input or the '--more--' prompt? Because those are the only two things that have a green color set in the config file.

**Comment by angstsmurf** [comment]
Yes, the "a" has the input style.

[Inform 7 source code](https://gist.github.com/angstsmurf/2d71eab0783f7d1e8425755161ad5e96).

**Comment by Tynach** [comment]
Huh, and input uses fixed width text. First time I've seen that.

I don't have Inform's IDE installed. Apparently inform7 is not yet open source or free software, though inform6 is, so inform6 is all that's available in the repositories.

**Comment by angstsmurf** [comment]
Well, I'm sure there are plenty of people out there who could translate that into Inform 6, but I'm afraid I'm not one of them.

Perhaps someone at the [infiction.org](https://intfiction.org) forum might be able to help.

**Comment by Tynach** [comment]
Nah, that's alright @angstsmurf. While I'll admit I have an interest in writing interactive fiction, I *really* don't like Inform's syntax. I feel like if I were to actually make my own interactive fiction game, I'd just find a way to compile Python or something like that into glulx bytecode.

Anyway, I remember trying to find the cause of a bug to do with selecting text backwards, and ran across this function:
https://github.com/garglk/garglk/blob/master/garglk/wintext.c#L363

It.. Seems to do a *lot* of different things, and it's an extremely large function. Since it's in charge of drawing all the text, I imagine handling font sizes and other styles might have to go in there.

I see some lines in that function where they use `gli_leading`, which is the number of pixels representing the height of one line of text (as set in the config file). For variable-height lines, that would probably need to be something stored in the line data itself instead.. And there'd potentially need to be a *lot* of refactoring.

This also ties into how Gargoyle currently treats points and pixels as the same thing, but that might not be valid if we want each line to start on a whole pixel but calculate the line's height front font data.

Speaking of which, we might need to do something like that for proper high-dpi support *any*way.
